Cargo.toml

```diff
@@ -63,6 +64,13 @@ preserve_order = ["indexmap"]
 # written back to a JSON string without loss of precision.
 arbitrary_precision = []
 
+# Deserialize floats with perfect accuracy and consistent rounding. This
+# changes the implementation for parsing both floats and integers. This
+# implementation is slower than the default and contains a lookup table that
+# increases artifact size. It does not affect serialization. This
+# feature has no effect if arbitrary_precision is enabled.
+perfect_float = ["lexical-core"]
```

Comment:
I would recommend using `perfect_float` by default. @dtolnay will ultimately decide, however. Correctness should always be favored by default, and performance at the cost of correctness should be opt-in, not opt-out.